### ISSUE TITLE
error: cannot copy tree 'build/scripts-3.8': not a directory

### ISSUE THREAD

**Issue by jachymb** [pypa/setuptools#2586]
Using python 3.8. Trying to build setuptools fails wih the following message on my system:

    error: cannot copy tree 'build/scripts-3.8': not a directory

simply creating an empty directory 'build/scripts-3.8' was apparently a successful workarond, but I feel the error should not have occured.

**Comment by jachymb** [comment]
Possibly related to #2550

**Comment by blue-devil** [comment]
I got the exact same issue, while making Pisi packages for PisiLinux.
So i have created an empty directoy `build/scripts-3.8` like you. Then i made the package successfully.

So is this the solution?

**Comment by mgorny** [comment]
We're hitting this when bootstrapping setuptools via the PEP517 entry point on a system where there is no setuptools yet.

Relevant bug report: https://bugs.gentoo.org/836670
Example build log: https://836670.bugs.gentoo.org/attachment.cgi?id=768544

**Comment by mgorny** [comment]
I'm not sure if OP's situation is the same as mine but the problem seems to stem from setuptools requiring entry points to set up its own commands. When bootstrapping, the entry points aren't present yet, so setuptools uses distutils' version of `install_scripts` which apparently doesn't work for building itself.

**Comment by mgorny** [comment]
Curious enough, I can't produce with 60.10.0, so in my case the problem is pretty recent. @abravalheri, any clue if pyproject.toml changes could have cause this?

**Comment by abravalheri** [comment]
> We're hitting this when bootstrapping setuptools via the PEP517 entry point on a system where there is no setuptools yet.

Hi @mgorny how are you guys doing that?
I am trying to follow the logs in the links, but I don't understand how setuptools is being built/installed.
(Sorry, I am not familiar with gentoo at all).

I tried the following, but I cannot see things going wrong:
```bash
> docker run --rm -it gentoo/stage3 /bin/bash


cd /tmp
wget https://www.python.org/ftp/python/3.8.11/Python-3.8.11.tar.xz
tar xf Python-3.8.11.tar.xz
cd Python-3.8.11
./configure --with-ensurepip=no
make
make install

# Make sure python3.8 is installed without setuptools
python3.8 -VV
# Python 3.8.11 (default, Apr  3 2022, 16:53:00)
# [GCC 11.2.1 20220115]
python3.8 -c 'import pip'
# ModuleNotFoundError: No module named 'pip'
python3.8 -c 'import setuptools'
# ModuleNotFoundError: No module named 'setuptools'

# Install a PEP 517 frontend without installing setuptools
cd /tmp
curl -sSL https://bootstrap.pypa.io/get-pip.py -o get-pip.py
python3.8 get-pip.py --no-setuptools --no-wheel
python3.8 -m pip install build
python3.8 -c 'import setuptools'
# ModuleNotFoundError: No module named 'setuptools'

# Build setuptools from source code
wget https://github.com/pypa/setuptools/archive/refs/tags/v61.3.1.tar.gz
tar xf v61.3.1.tar.gz
cd setuptools-61.3.1
python3.8 -m build
# Successfully built setuptools-61.3.1.post20220403.tar.gz and setuptools-61.3.1.post20220403-py3-none-any.whl

# install setuptools
python3.8 -m pip install dist/setuptools-61.3.1.post20220403-py3-none-any.whl
# Successfully installed setuptools-61.3.1.post20220403

python3.8 -c 'import setuptools; print(setuptools.__path__, setuptools.__version__)'
# ['/tmp/setuptools-61.3.1/setuptools'] 61.3.1.post20220403
```

This might be related to the fact that there is another installation of `python/setuptools` in the container, however, the folder were that other copy of setuptools is installed should not be available for `python3.8`:
```bash
python3 -c 'import setuptools; print(setuptools.__path__, setuptools.__version__)'
# ['/usr/lib/python3.9/site-packages/setuptools'] 60.9.2
```

**Comment by mgorny** [comment]
I think pip might be "breaking" your reproducer. I managed to reproduce it with the following commands inside the Docker shell:

```
rm -r /usr/lib/python3.9/site-packages/setuptools*
cd /tmp
wget https://github.com/pypa/setuptools/archive/refs/tags/v61.3.1.tar.gz
tar xf v61.3.1.tar.gz
cd setuptools-61.3.1
python3.9 -c 'import setuptools.build_meta; setuptools.build_meta.build_wheel("/tmp")'
```

**Comment by abravalheri** [comment]
I am not 100% sure here, but I suspect that if you are going to call `setuptools` directly without a frontend you might need to run `egg_info` before hand. Let me try again...

**Comment by abravalheri** [comment]
@mgorny, the same problem happens with 60.10.0:
 
```bash
docker run --rm -it gentoo/stage3 /bin/bash
rm -r /usr/lib/python3.9/site-packages/setuptools*
cd /tmp
wget https://github.com/pypa/setuptools/archive/refs/tags/v60.10.0.tar.gz
tar xf v60.10.0.tar.gz
cd setuptools-60.10.0/
python3.9 -c 'import setuptools.build_meta; setuptools.build_meta.build_wheel("/tmp")'
# ...
# error: cannot copy tree 'build/scripts-3.9': not a directory
```

**Comment by abravalheri** [comment]
Yes, so running `egg_info` before hand seems to solve the problem (which is not a real solution since running `setup.py` directly is deprecated). So I guess this is the same already existing problem described here previously.

Adding `install_scripts` to `bootstrap.egg-info/entry_points.txt` might help...

**Comment by mgorny** [comment]
Oh, I'm sorry, it just occurred to me that I've switched 61.3.0 from pypi sdist to GitHub archive because of the missing test files. Switching back to the sdist archive indeed resolves that. I'm sorry for the noise.

Though I suppose we now know what might have caused the original report.
